packages/flutter/lib/src/widgets/image_filter.dart

```diff
@@ -23,20 +28,34 @@ class ImageFiltered extends SingleChildRenderObjectWidget {
   /// The [imageFilter] must not be null.
   const ImageFiltered({
     Key? key,
-    required this.imageFilter,
+    this.imageFilter,
+    this.imageFilterCallback,
     Widget? child,
-  }) : assert(imageFilter != null),
+  }) : assert(imageFilter != null || imageFilterCallback != null,
+              'One of imageFilter or imageFilterCallback should be non-null'),
+       assert(imageFilter == null || imageFilterCallback == null,
+              'Only one of imageFilter or imageFilterCallback should be non-null'),
        super(key: key, child: child);
 
   /// The image filter to apply to the child of this widget.
-  final ImageFilter imageFilter;
+  final ImageFilter? imageFilter;
+
+  /// The callback
```

Comment:
these docs are, uh, anemic :-)

This should talk about what the arguments to the callback are, when the callback is called, what to do if the callback wants to change what it returns but nothing else has changed (e.g. if you want to animate the shader), what the ownership model for the returned value is, etc. (See the style guide for writing prompts for docs.)

Reply:
This case was a complete 100% oops. I must have gotten distracted before I finished it. I'll do a pass and beef them up.

Reply:
Lots and lots of beefed up docs in the next revision.

Reply:
Docs are completely rewritten in the latest commit.